scripts/docker-cli-monitor.ts

```diff
@@ -0,0 +1,72 @@
+import path from 'path';
+
+import { readDependencyVersions, getOctokit, RancherDesktopRepository, IssueOrPullRequest } from 'scripts/lib/dependencies';
+
+const GITHUB_OWNER = process.env.GITHUB_REPOSITORY?.split('/')[0] || 'rancher-sandbox';
+const GITHUB_REPO = process.env.GITHUB_REPOSITORY?.split('/')[1] || 'rancher-desktop';
+const DOCKER_CLI_OWNER = process.env.DOCKER_CLI_OWNER || 'docker';
+const DOCKER_CLI_REPO = process.env.DOCKER_CLI_REPO || 'cli';
+const TAG_REGEX = /^v[0-9]+\.[0-9]+\.[0-9]+$/;
+const SCRIPT_NAME = 'docker-cli-monitor';
+const mainRepo = new RancherDesktopRepository(GITHUB_OWNER, GITHUB_REPO);
+
+async function getLatestDockerCliVersion(): Promise<string> {
+  const result = await getOctokit().rest.repos.listTags({ owner: DOCKER_CLI_OWNER, repo: DOCKER_CLI_REPO });
+  const tags = result.data;
+  const latestTag = tags.find(tag => TAG_REGEX.test(tag.name));
```

Comment:
It seems that docker versions are super weird. I thought they looked like semver, but that doesn't explain the middle set of numbers (we went from 18.09.xx to 19.03.xx to 20.10.xx). My best guess is that the first two numbers are the year and month of the last "major" version release, with the last set of numbers being the "patch" release.

Either way, `semver` is choking on numbers like `19.03.15`, I think because of the 0 in `03`. And we can't sort by alphabetical order, because then `19.03.2` comes before `19.03.10`. Funny enough, the array returned from github seems to be sorted correctly:
```
[
  'v20.10.23', 'v20.10.22', 'v20.10.21', 'v20.10.20',
  'v20.10.19', 'v20.10.18', 'v20.10.17', 'v20.10.16',
  'v20.10.15', 'v20.10.14', 'v20.10.13', 'v20.10.12',
  'v20.10.11', 'v20.10.10', 'v20.10.9',  'v20.10.8',
  'v20.10.7',  'v20.10.6',  'v20.10.5',  'v20.10.4',
  'v20.10.3',  'v20.10.2',  'v20.10.1',  'v20.10.0',
  'v19.03.15', 'v19.03.14', 'v19.03.13', 'v19.03.12',
  'v19.03.11', 'v19.03.10', 'v19.03.9',  'v19.03.8',
  'v19.03.7',  'v19.03.6',  'v19.03.5',  'v19.03.4',
  'v19.03.3',  'v19.03.2',  'v19.03.1',  'v19.03.0',
  'v18.09.9',  'v18.09.8',  'v18.09.7',  'v18.09.6',
  'v18.09.5',  'v18.09.4',  'v18.09.3',  'v18.09.2',
  'v18.09.1',  'v18.09.0'
]
```
Note that I removed the beta and rc version numbers from the above. It seems like github knows how docker versions work, and yet I can't find anywhere it's documented. I'll take more of a look tomorrow.